### PR TITLE
Added selection of currency for account to consentUI (dedicated consent flow)

### PR DESCRIPTION
**Pull request by tsehelnyk**
Fixed bug in consentUI when creating dedicated consent with several accounts (displaying an incorrect list of accounts)

### CODE REVIEW

**Review by valb3r — COMMENTED** [review]
Please add backend test (or Selenium test) that validates selected currency is in use

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java [hunk 1]
```diff
@@ -192,18 +195,25 @@ public void open_banking_can_not_read_anton_brueckner_account_data_using_consent
...
+        return open_banking_can_read_user_account_data_using_consent_bound_to_service_session(user, validateResourceId, "EUR");
+    }
+
+    @SneakyThrows
+    public SELF open_banking_can_read_user_account_data_using_consent_bound_to_service_session(
+            String user, boolean validateResourceId, String currency
     ) {
         ExtractableResponse<Response> response = withAccountsHeaders(user)
-                     .header(SERVICE_SESSION_ID, serviceSessionId)
+                .header(SERVICE_SESSION_ID, serviceSessionId)
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java:207]: Please keep old formatting

File: smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java [hunk 2]
```diff
@@ -97,16 +111,30 @@ private SELF create_account_for_user_in_sandbox_tpp_management(String auth, Stri
...
+    }
+
+    private SELF create_account_for_user_in_sandbox_tpp_management(String auth, String userId, String accountBody) {
         RestAssured
                 .given()
-                    .contentType(MediaType.APPLICATION_JSON_VALUE)
-                    .header(TPP_MANAGEMENT_AUTH_HEADER, auth)
-                    .queryParam(TPP_MANAGEMENT_USER_ID_QUERY, userId)
-                    .body(body)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
```
Comment (valb3r) [inline smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java:130]: Please follow old formatting

File: smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java [hunk 3]
```diff
@@ -130,16 +158,24 @@ private String get_account_id_by_iban_in_sandbox_tpp_management(String auth, Str
...
+        return deposit_to_sandbox_tpp_management_user_account(auth, accountId, body);
+    }
 
+    private SELF deposit_to_sandbox_tpp_management_user_account(String auth, String accountId, String body) {
         RestAssured
                 .given()
-                    .contentType(MediaType.APPLICATION_JSON_VALUE)
-                    .header(TPP_MANAGEMENT_AUTH_HEADER, auth)
-                    .body(body)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
```
Comment (valb3r) [inline smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java:172]: Please follow old formatting

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java (hunk 1 above)
Comment (tsehelnyk) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationResult.java]: fixed

File: smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java (hunk 2 above)
Comment (tsehelnyk) [reply smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java]: fixed

File: smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java (hunk 3 above)
Comment (tsehelnyk) [reply smoke-tests/src/test/java/de/adorsys/opba/smoketests/steps/SmokeSandboxServers.java]: fixed

**Review by valb3r — APPROVED** [review]

**Pull request merged**